File: BankingApp_WDP.API/DTOs/Notifications/NotificationResponse.cs
using BankingApp.Models;

namespace BankingApp.DTOs.Notifications;

public class NotificationResponse
{
    public int Id { get; set; }
    public string Title { get; set; } = null!;
    public string Message { get; set; } = null!;
    public string Type { get; set; } = null!;
    public bool IsRead { get; set; }
    public DateTime CreatedAt { get; set; }

    public static NotificationResponse FromNotification(Notification n) => new()
    {
        Id = n.Id,
        Title = n.Title,
        Message = n.Message,
        Type = n.Type.ToString(),
        IsRead = n.IsRead,
        CreatedAt = n.CreatedAt
    };
}

File: BankingApp_WDP.APP/app/src/main/java/com/example/bankingapp/data/model/notification/NotificationResponse.kt
package com.example.bankingapp.data.model.notification

import com.google.gson.annotations.SerializedName

data class NotificationResponse(
    @SerializedName("id")        val id: String,
    @SerializedName("type")      val type: String,
    @SerializedName("title")     val title: String,
    @SerializedName("message")   val message: String,
    @SerializedName("isRead")    val isRead: Boolean,
    @SerializedName("createdAt") val createdAt: String
)
